// hcal/esm2015/app/notification/notification-content/notification-content.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class NotificationContentComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
NotificationContentComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-notification-content',
                template: `<main [class]="'content-notification float-left '+contentSize">
  <div [class]="'col-12 float-left notification-head bg-'+notificationType">
    <h3 class="h3">{{ notificationTitle }}</h3>
  </div>
  <div [class]="'col-12 offset-0 notification-body float-left bg-'+notificationType">
    <p class="text">{{ notificationText }}</p>
    <div class="col-12 float-right">
      <button [class]="'button col-2 float-right button-'+notificationType">Ok</button>
    </div>
  </div>
</main>
`,
                styles: [`.notification-head{float:left;height:80px;border-top-left-radius:15px}.notification-head .h3{padding:30px 10px;font-family:Poppins,sans-serif;color:#fff}.notification-body{height:auto;border-bottom-right-radius:15px}.notification-body .text{margin-top:15px;padding:10px;line-height:30px;font-family:Poppins,sans-serif;text-align:justify;color:#fff}.notification-body .button{margin-top:100px;margin-bottom:10px;margin-right:10px}`]
            },] },
];
/** @nocollapse */
NotificationContentComponent.ctorParameters = () => [];
NotificationContentComponent.propDecorators = {
    notificationText: [{ type: Input }],
    notificationTitle: [{ type: Input }],
    notificationType: [{ type: Input }],
    contentSize: [{ type: Input }]
};
function NotificationContentComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    NotificationContentComponent.prototype.notificationText;
    /** @type {?} */
    NotificationContentComponent.prototype.notificationTitle;
    /** @type {?} */
    NotificationContentComponent.prototype.notificationType;
    /** @type {?} */
    NotificationContentComponent.prototype.contentSize;
}
