// hcal/esm5/app/form/input-base/input-base.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
var InputBaseComponent = /** @class */ (function () {
    function InputBaseComponent() {
        this.elementType = 'text';
        this.readonly = false;
        this.elementPlaceholder = '';
        this.enableTooltip = true;
    }
    /**
     * @return {?}
     */
    InputBaseComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    InputBaseComponent.decorators = [
        { type: Component, args: [{
                    selector: 'hcal-input-base',
                    template: "<div [class]=\"'form-group position-relative '+contentStyle\">\n  <label [attr.for]=\"elementId\" class=\"control-label\">{{ elementLabel }}</label>\n  <input [type]=\"elementType\" [class]=\"className\" name=\"elementName\" [id]=\"elementId\" [placeholder]=\"elementPlaceholder\" [position]=\"placementTooltip\"\n         [tooltipHcal]=\"textTooltip\" [delay]=\"delay\" *ngIf=\"enableTooltip\" ngModel [required]=\"enableRequired\">\n  <input *ngIf=\"!enableTooltip\" [type]=\"elementType\" [class]=\"className\" ngModel name=\"elementName\" [id]=\"elementId\" [placeholder]=\"elementPlaceholder\"\n         [required]=\"enableRequired\"\n         [(ngModel)]=\"elementNgModel\">\n</div>\n",
                    styles: [""]
                },] },
    ];
    /** @nocollapse */
    InputBaseComponent.ctorParameters = function () { return []; };
    InputBaseComponent.propDecorators = {
        elementId: [{ type: Input }],
        elementType: [{ type: Input }],
        elementLabel: [{ type: Input }],
        readonly: [{ type: Input }],
        maxLength: [{ type: Input }],
        elementNgModel: [{ type: Input }],
        elementPlaceholder: [{ type: Input }],
        elementName: [{ type: Input }],
        className: [{ type: Input }],
        textTooltip: [{ type: Input }],
        contentStyle: [{ type: Input }],
        placementTooltip: [{ type: Input }],
        delay: [{ type: Input }],
        enableTooltip: [{ type: Input }],
        enableRequired: [{ type: Input }]
    };
    return InputBaseComponent;
}());
export { InputBaseComponent };
function InputBaseComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    InputBaseComponent.prototype.elementId;
    /** @type {?} */
    InputBaseComponent.prototype.elementType;
    /** @type {?} */
    InputBaseComponent.prototype.elementLabel;
    /** @type {?} */
    InputBaseComponent.prototype.readonly;
    /** @type {?} */
    InputBaseComponent.prototype.maxLength;
    /** @type {?} */
    InputBaseComponent.prototype.elementNgModel;
    /** @type {?} */
    InputBaseComponent.prototype.elementPlaceholder;
    /** @type {?} */
    InputBaseComponent.prototype.elementName;
    /** @type {?} */
    InputBaseComponent.prototype.className;
    /** @type {?} */
    InputBaseComponent.prototype.textTooltip;
    /** @type {?} */
    InputBaseComponent.prototype.contentStyle;
    /** @type {?} */
    InputBaseComponent.prototype.placementTooltip;
    /** @type {?} */
    InputBaseComponent.prototype.delay;
    /** @type {?} */
    InputBaseComponent.prototype.enableTooltip;
    /** @type {?} */
    InputBaseComponent.prototype.enableRequired;
}
