// hcal/esm2015/app/form/select-base/select-base.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { InputBaseComponent } from '../input-base/input-base.component';
export class SelectBaseComponent extends InputBaseComponent {
    constructor() {
        super();
        this.options = [];
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
SelectBaseComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-select-base',
                template: `<div [class]="'form-group '+contentStyle">
  <label [attr.for]="elementId" class="control-label">{{ elementLabel }}</label>
  <select [name]="elementName" [id]="elementId" [class]="className">
    <option value="" *ngFor="let option of options">{{ option.object.nome }}</option>
  </select>
</div>
`,
                styles: [``]
            },] },
];
/** @nocollapse */
SelectBaseComponent.ctorParameters = () => [];
SelectBaseComponent.propDecorators = {
    options: [{ type: Input }]
};
function SelectBaseComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    SelectBaseComponent.prototype.options;
}
