// hcal/esm5/app/tooltip.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, ElementRef, HostListener, Input, Renderer2 } from '@angular/core';
var TooltipDirective = /** @class */ (function () {
    function TooltipDirective(_elementRef, _renderRef) {
        this._elementRef = _elementRef;
        this._renderRef = _renderRef;
        this.delay = 1000;
        this.isLabel = true;
        this.element = this._renderRef.selectRootElement(this._elementRef.nativeElement);
    }
    /**
     * @param {?} position
     * @return {?}
     */
    TooltipDirective.prototype.create = /**
     * @param {?} position
     * @return {?}
     */
    function (position) {
        this.parent = this._renderRef.parentNode(this.element);
        this.tooltipSpan = this._renderRef.createElement('span');
        this.tooltipText = this._renderRef.createText(this.tooltipHcal);
        this._renderRef.appendChild(this.tooltipSpan, this.tooltipText);
        switch (position) {
            case 'top':
                this.insertBefore();
                break;
            case 'bottom':
                this.insertAfter();
                break;
            default:
                this.insertAfter();
                break;
        }
    };
    /**
     * @return {?}
     */
    TooltipDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.create(this.position);
        this._renderRef.addClass(this.tooltipSpan, 'off-tooltipHcal');
    };
    /**
     * @param {?} delay
     * @param {?} _renderRef
     * @param {?} tooltipSpan
     * @param {?} position
     * @param {?} element
     * @param {?} isLabel
     * @return {?}
     */
    TooltipDirective.prototype.setTooltipInterval = /**
     * @param {?} delay
     * @param {?} _renderRef
     * @param {?} tooltipSpan
     * @param {?} position
     * @param {?} element
     * @param {?} isLabel
     * @return {?}
     */
    function (delay, _renderRef, tooltipSpan, position, element, isLabel) {
        this.onDelay = setTimeout(function () {
            _renderRef.removeClass(tooltipSpan, 'off-tooltipHcal');
            _renderRef.addClass(tooltipSpan, 'on-tooltipHcal');
            _renderRef.addClass(tooltipSpan, position);
            /** @type {?} */
            var marginTop = '';
            if (isLabel) {
                marginTop = "calc(75% - " + element.offsetHeight / 2 + "px)";
            }
            else {
                marginTop = "calc(50% - " + element.offsetHeight / 2 + "px)";
            }
            /** @type {?} */
            var marginLeft = "calc(50% - " + tooltipSpan.offsetWidth / 2 + "px";
            if (position === 'right' || position === 'left') {
                _renderRef.setStyle(tooltipSpan, 'top', marginTop);
                console.log(tooltipSpan.offsetHeight);
            }
            else {
                _renderRef.setStyle(tooltipSpan, 'left', marginLeft);
            }
        }, delay, false);
    };
    /**
     * @return {?}
     */
    TooltipDirective.prototype.onMouseOver = /**
     * @return {?}
     */
    function () {
        this.setTooltipInterval(this.delay, this._renderRef, this.tooltipSpan, this.position, this.element, this.isLabel);
    };
    /**
     * @return {?}
     */
    TooltipDirective.prototype.outMouseOver = /**
     * @return {?}
     */
    function () {
        clearTimeout(this.onDelay);
        this._renderRef.removeClass(this.tooltipSpan, this.position);
        this._renderRef.removeClass(this.tooltipSpan, 'on-tooltipHcal');
        this._renderRef.addClass(this.tooltipSpan, 'off-tooltipHcal');
    };
    /**
     * @return {?}
     */
    TooltipDirective.prototype.insertBefore = /**
     * @return {?}
     */
    function () {
        this._renderRef.insertBefore(this.parent, this.tooltipSpan, this.element);
    };
    /**
     * @return {?}
     */
    TooltipDirective.prototype.insertAfter = /**
     * @return {?}
     */
    function () {
        this._renderRef.appendChild(this.parent, this.tooltipSpan);
    };
    TooltipDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[tooltipHcal]'
                },] },
    ];
    /** @nocollapse */
    TooltipDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 }
    ]; };
    TooltipDirective.propDecorators = {
        tooltipHcal: [{ type: Input }],
        position: [{ type: Input }],
        delay: [{ type: Input }],
        isLabel: [{ type: Input }],
        onMouseOver: [{ type: HostListener, args: ['mouseover',] }],
        outMouseOver: [{ type: HostListener, args: ['mouseleave',] }]
    };
    return TooltipDirective;
}());
export { TooltipDirective };
function TooltipDirective_tsickle_Closure_declarations() {
    /** @type {?} */
    TooltipDirective.prototype.tooltipHcal;
    /** @type {?} */
    TooltipDirective.prototype.position;
    /** @type {?} */
    TooltipDirective.prototype.delay;
    /** @type {?} */
    TooltipDirective.prototype.isLabel;
    /** @type {?} */
    TooltipDirective.prototype.element;
    /** @type {?} */
    TooltipDirective.prototype.tooltipSpan;
    /** @type {?} */
    TooltipDirective.prototype.tooltipText;
    /** @type {?} */
    TooltipDirective.prototype.parent;
    /** @type {?} */
    TooltipDirective.prototype.onDelay;
    /** @type {?} */
    TooltipDirective.prototype._elementRef;
    /** @type {?} */
    TooltipDirective.prototype._renderRef;
}
