// hcal/esm2015/app/form/custom-dual-list/custom-dual-list.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { DualListComponent } from 'angular-dual-listbox';
export class CustomDualListComponent extends DualListComponent {
    constructor() {
        super(...arguments);
        this.format = {
            add: '', remove: '', all: 'Todos', none: 'Nenhum',
            direction: DualListComponent.LTR, draggable: true, locale: 'br'
        };
        this.fieldDestination = [];
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
CustomDualListComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-custom-dual-list',
                template: `<dual-list [source]="data" [destination]="fieldDestination" [format]="format"></dual-list>
`,
                styles: [``]
            },] },
];
CustomDualListComponent.propDecorators = {
    data: [{ type: Input }]
};
function CustomDualListComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    CustomDualListComponent.prototype.format;
    /** @type {?} */
    CustomDualListComponent.prototype.data;
    /** @type {?} */
    CustomDualListComponent.prototype.fieldDestination;
}
