// hcal/esm2015/app/main/nav/nav.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class NavComponent {
    constructor() {
        this.hover = false;
        this.hiddenCog = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
NavComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-nav',
                template: `<nav class="menu-container main-menu" main-menu>
  <ul class="menu-content">
    <li class="menu-item">
      <ng-content select="[dropdown]"></ng-content>
    </li>
    <li class="menu-item" [hidden]="hiddenCog">
      <a routerLink="" class="icon">
        <fa-icon class="icon-image cogs-icon"
                 [icon]="['fas', 'cog']"
                 (mouseover)="hover=true"
                 (mouseleave)="hover=false"
                 [spin]="hover"></fa-icon>
      </a>
    </li>
    <li class="menu-item">
      <a routerLink="" class="icon">
        <fa-icon [icon]="['fas', 'sign-out-alt']" class="icon-image"></fa-icon>
      </a>
    </li>
  </ul>
</nav>
`,
                styles: [`.menu-container{margin-top:4px;margin-right:3%;float:right}.menu-container .menu-content{margin-top:3px;width:auto;padding:0;float:left}.menu-container .menu-content .menu-item{float:left;margin-top:27px;margin-right:15px}.icon-image{margin-right:12px;display:inline-block;color:#fff;font-size:1.2em}`]
            },] },
];
/** @nocollapse */
NavComponent.ctorParameters = () => [];
NavComponent.propDecorators = {
    hiddenCog: [{ type: Input }]
};
function NavComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    NavComponent.prototype.hover;
    /** @type {?} */
    NavComponent.prototype.hiddenCog;
}
