// hcal/esm2015/app/form/input-base/input-base.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class InputBaseComponent {
    constructor() {
        this.elementType = 'text';
        this.readonly = false;
        this.elementPlaceholder = '';
        this.enableTooltip = true;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
InputBaseComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-input-base',
                template: `<div [class]="'form-group position-relative '+contentStyle">
  <label [attr.for]="elementId" class="control-label">{{ elementLabel }}</label>
  <input [type]="elementType" [class]="className" name="elementName" [id]="elementId" [placeholder]="elementPlaceholder" [position]="placementTooltip"
         [tooltipHcal]="textTooltip" [delay]="delay" *ngIf="enableTooltip" ngModel [required]="enableRequired">
  <input *ngIf="!enableTooltip" [type]="elementType" [class]="className" ngModel name="elementName" [id]="elementId" [placeholder]="elementPlaceholder"
         [required]="enableRequired"
         [(ngModel)]="elementNgModel">
</div>
`,
                styles: [``]
            },] },
];
/** @nocollapse */
InputBaseComponent.ctorParameters = () => [];
InputBaseComponent.propDecorators = {
    elementId: [{ type: Input }],
    elementType: [{ type: Input }],
    elementLabel: [{ type: Input }],
    readonly: [{ type: Input }],
    maxLength: [{ type: Input }],
    elementNgModel: [{ type: Input }],
    elementPlaceholder: [{ type: Input }],
    elementName: [{ type: Input }],
    className: [{ type: Input }],
    textTooltip: [{ type: Input }],
    contentStyle: [{ type: Input }],
    placementTooltip: [{ type: Input }],
    delay: [{ type: Input }],
    enableTooltip: [{ type: Input }],
    enableRequired: [{ type: Input }]
};
function InputBaseComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    InputBaseComponent.prototype.elementId;
    /** @type {?} */
    InputBaseComponent.prototype.elementType;
    /** @type {?} */
    InputBaseComponent.prototype.elementLabel;
    /** @type {?} */
    InputBaseComponent.prototype.readonly;
    /** @type {?} */
    InputBaseComponent.prototype.maxLength;
    /** @type {?} */
    InputBaseComponent.prototype.elementNgModel;
    /** @type {?} */
    InputBaseComponent.prototype.elementPlaceholder;
    /** @type {?} */
    InputBaseComponent.prototype.elementName;
    /** @type {?} */
    InputBaseComponent.prototype.className;
    /** @type {?} */
    InputBaseComponent.prototype.textTooltip;
    /** @type {?} */
    InputBaseComponent.prototype.contentStyle;
    /** @type {?} */
    InputBaseComponent.prototype.placementTooltip;
    /** @type {?} */
    InputBaseComponent.prototype.delay;
    /** @type {?} */
    InputBaseComponent.prototype.enableTooltip;
    /** @type {?} */
    InputBaseComponent.prototype.enableRequired;
}
