// hcal/esm2015/app/report/report.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component } from '@angular/core';
import { Chart } from 'angular-highcharts';
export class ReportComponent {
    constructor() {
        this.chart = new Chart({
            chart: {
                type: 'pie'
            },
            title: {
                text: 'Graph',
            },
            credits: {
                enabled: false
            },
            series: [{
                    name: 'Pie 1',
                    data: [1, 2, 3]
                }]
        });
        this.componentName = this.constructor.name;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
ReportComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-report',
                template: `<h3 class="title-component">{{ componentName }}</h3>
<div class="content graph-container col-4 float-left">
  <div [chart]="chart" class="col-12 chart-content float-left"></div>
</div>
<div class="col-6 offset-1 float-left">
  <ng-content select="[table-report]"></ng-content>
</div>
`,
                styles: [`.graph-container{height:calc(100% - 140px)!important}`]
            },] },
];
/** @nocollapse */
ReportComponent.ctorParameters = () => [];
function ReportComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    ReportComponent.prototype.componentName;
    /** @type {?} */
    ReportComponent.prototype.chart;
}
