// hcal/esm2015/app/aside-bar/aside-bar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class AsideBarComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
AsideBarComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-aside-bar',
                template: `<main class="container">
  <header class="top">
    <img id="logo" [src]="urlLogo" alt="Logo - Hospital do Coração de Alagoas">
  </header>
  <div class="menu-container">
    <ul class="menu-content">
      <ng-content select="[main-menu]"></ng-content>
    </ul>
  </div>
  <footer class="bottom"></footer>
</main>
`,
                styles: [`.container{position:fixed;width:150px;height:100%;float:left;z-index:initial;background-color:#f2f2f2}.container .top{width:100%;height:90px;background:#00aba5}.container .top #logo{width:70%;margin-top:5px;margin-left:16px;float:left}.container .menu-container{height:80%;z-index:5;position:relative}.container .menu-container .menu-content{position:relative;width:100%;margin:0;padding:0}.container .bottom{width:calc(100% - 140px);height:25px;position:fixed;bottom:0;background:#00aba5}`]
            },] },
];
/** @nocollapse */
AsideBarComponent.ctorParameters = () => [];
AsideBarComponent.propDecorators = {
    urlLogo: [{ type: Input }]
};
function AsideBarComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    AsideBarComponent.prototype.urlLogo;
}
