// hcal/esm2015/app/aside-bar/link/link.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class LinkComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
LinkComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-link',
                template: `<li class="menu-item" routerLinkActive="active">
  <a [routerLink]="routerLinkName" class="full-link">
    <fa-icon [icon]="['fas', iconLink]" class="icon-aside"></fa-icon>
    <h3 class="title-link">{{ titleLink }}</h3>
  </a>
</li>
`,
                styles: [`.menu-item{margin-top:8px;margin-bottom:-3px;height:auto}.menu-item :hover{background-color:rgba(0,171,165,.45);transition:.7s}.menu-item .full-link{display:block;margin-left:-7.5px;text-decoration:none;text-align:center;font-family:"Roboto Medium",sans-serif;color:#00aba5}.menu-item .full-link:hover{color:#fff;background:rgba(0,171,165,.45)}.menu-item .full-link .icon-aside,.menu-item .full-link .title-link{display:block}.menu-item .full-link .icon-aside:hover,.menu-item .full-link .title-link:hover{background:0 0}.menu-item .full-link .icon-aside{margin-bottom:-7px;font-size:3em}.menu-item .full-link .title-link{padding-bottom:5px;width:auto;font-size:.9em;letter-spacing:.9px;font-weight:400}.active{background:rgba(0,171,165,.45)}.active .full-link{color:#fff}`]
            },] },
];
/** @nocollapse */
LinkComponent.ctorParameters = () => [];
LinkComponent.propDecorators = {
    titleLink: [{ type: Input }],
    routerLinkName: [{ type: Input }],
    iconLink: [{ type: Input }]
};
function LinkComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    LinkComponent.prototype.titleLink;
    /** @type {?} */
    LinkComponent.prototype.routerLinkName;
    /** @type {?} */
    LinkComponent.prototype.iconLink;
}
