// hcal/esm2015/app/table/table.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class TableComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
TableComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-table',
                template: `<div class="table-container">
  <table class="table-default">
    <caption class="tname">{{ tableName }}</caption>
    <thead class="thead-container">
    <tr class="trow-head">
      <ng-content select="[thead-content]"></ng-content>
    </tr>
    </thead>
    <tbody class="tbody-container">
      <ng-content></ng-content>
    </tbody>
  </table>
</div>
`,
                styles: [`.table-default{float:left;max-width:100%;min-width:100%;border-collapse:collapse;cursor:default}.table-default .tname{margin-bottom:10px;font-family:Roboto,sans-serif;text-align:left;font-size:1em;color:#2b517e}.table-default .thead-container{box-shadow:0 5px 12px -7px #000}.table-default .thead-default{padding:10px 40px;color:#fff;background:#2b517e;font-family:Roboto,sans-serif;text-transform:uppercase}.table-default .thead-default:first-child{border-top-left-radius:8px}.table-default .thead-default:last-child{border-top-right-radius:8px}.table-default .tbody-container .trow-content{color:#2b517e;font-family:Roboto,sans-serif;font-weight:300;text-align:center}.table-default .tbody-container .trow-content:nth-child(even){background:rgba(41,81,126,.1)}.table-default .tbody-container .trow-content:hover{cursor:pointer;background:rgba(41,81,126,.2);transition:.5s}.table-default .tbody-container .trow-content:last-child .tdata-default:first-child{border-bottom-left-radius:8px}.table-default .tbody-container .trow-content:last-child .tdata-default:last-child{border-bottom-right-radius:8px}.table-default .icon-table{color:#fff;font-size:1.2em}.table-default,.tdata-default,.thead-default,.trow-content{padding:10px}`]
            },] },
];
/** @nocollapse */
TableComponent.ctorParameters = () => [];
TableComponent.propDecorators = {
    tableName: [{ type: Input }]
};
function TableComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    TableComponent.prototype.tableName;
}
