// hcal/esm2015/app/main/main.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class MainComponent {
    constructor() {
        this.title = 'System Name';
        this.credits = 'TECNOLOGIA DA INFORMAÇÃO - HCAL';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
MainComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-main',
                template: `<main class="container">
  <header class="top">
    <h2 class="title-header">{{ title }}</h2>
    <ng-content select="[nav-main]"></ng-content>
  </header>
  <section class="medium">
    <ng-content select="[route]"></ng-content>
  </section>
  <footer class="bottom">
    <p id="footer-ti">{{ credits }}</p>
  </footer>
</main>
`,
                styles: [`.container{position:relative;width:calc(100% - 140px);height:100%;float:left;display:block;margin-left:140px;z-index:1;box-shadow:-5px 0 15px -7px;border-top-left-radius:10px;border-bottom-left-radius:10px}.container .top{width:100%;height:90px;background:#4dc2b3;border-top-left-radius:10px}.container .top .title-header{float:left;margin-top:29px;margin-left:25px;font-family:Roboto,serif;color:#fff;font-weight:500;cursor:default;text-transform:uppercase}.medium{height:calc(100% - 150px);padding:18px 50px;background:#f2f2f2;border-bottom-left-radius:20px}.bottom{width:calc(100% - 140px);height:25px;position:fixed;bottom:0;background:#4dc2b3;border-bottom-left-radius:10px}#footer-ti{margin-top:4px;text-align:center;font-family:Roboto,serif;color:#fff;font-size:13px;letter-spacing:3px;font-weight:500;cursor:default}`]
            },] },
];
/** @nocollapse */
MainComponent.ctorParameters = () => [];
MainComponent.propDecorators = {
    title: [{ type: Input }]
};
function MainComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    MainComponent.prototype.title;
    /** @type {?} */
    MainComponent.prototype.credits;
}
