// hcal/esm2015/app/notification/notification-dropdown/notification-dropdown.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class NotificationDropdownComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
NotificationDropdownComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-notification-dropdown',
                template: `<div dropdown>
  <ng-content select="[dropdownToggle]"></ng-content>
  <ul class="dropdown-menu notification-dropdown" dropdownMenu>
    <li class="notification-item">
      <div [class]="'container-notification-icon float-left bg-'+typeNotificationDropdown">
        <fa-icon [icon]="['fas', 'check']" class="notification-icon"></fa-icon>
      </div>
      <div [class]="'container-notification-content float-left bg-light-'+typeNotificationDropdown">
        <h3 class="notification-title">{{ titleNotificationDropdown }}</h3>
        <p class="content-notification">{{ textNotificationDropdown }}</p>
        <div class="notification-content-action-dropdown float-right">
          <button [class]="'button notification-action-dropdown button-'+typeNotificationDropdown">Read</button>
          <button [class]="'button notification-action-dropdown button-'+typeNotificationDropdown">Ok</button>
        </div>
      </div>
    </li>
  </ul>
</div>
`,
                styles: [`.notification-dropdown{width:400px;display:inline-block;float:left;height:auto;padding-top:10px;position:absolute;border-radius:10px;cursor:pointer;box-shadow:2px 2px 15px #999;transition:.3s;background:#fff}.notification-dropdown .notification-item{margin-bottom:10px;height:auto;float:left;border-radius:10px}.notification-dropdown .notification-item:hover{box-shadow:2px 2px 15px #999;transition:.3s}.notification-dropdown .notification-item .container-notification-icon{float:left;display:inline-block;min-width:80px;padding:20px;text-align:center;border-top-left-radius:10px;border-bottom-left-radius:10px}.notification-dropdown .notification-item .notification-icon{line-height:5px;font-size:4em;color:#fff}.notification-dropdown .notification-item .container-notification-content{float:left;display:inline-block;height:104px;min-width:250px;max-width:250px;border-top-right-radius:10px;border-bottom-right-radius:10px}.notification-dropdown .notification-item .container-notification-content .notification-title{padding-top:5px;padding-left:8px;margin-bottom:4px}.notification-dropdown .notification-item .container-notification-content .content-notification,.notification-dropdown .notification-item .container-notification-content .notification-title{color:#fff;font-family:Roboto,sans-serif;letter-spacing:1px}.notification-dropdown .notification-item .container-notification-content .content-notification{padding-left:10px;padding-right:10px;font-size:.9em;text-align:justify}.notification-dropdown .notification-item .container-notification-content .notification-content-action-dropdown{margin-top:3px;padding-left:8px}.notification-dropdown .notification-item .container-notification-content .notification-content-action-dropdown .notification-action-dropdown{border-radius:8px;margin-right:5px;font-weight:700;color:#fff;border-bottom:none}.dropdown-container{min-width:150px;max-width:370px;padding:20px;list-style-position:outside;border-radius:4px;box-shadow:2px 2px 15px #575756}.dropdown-container .dropdown-notification-item .dropdown-notification-icon{line-height:5px;font-size:4em;color:#fff}.dropdown-container .dropdown-item{margin:0;transition:.3s;display:block}.dropdown-container .dropdown-item:hover{background:#4dc2b3;box-shadow:2px 2px 15px #999;cursor:pointer;transition:.3s;border-radius:inherit}.dropdown-container .dropdown-item .dropdown-link{width:100%;padding:10px 0 10px 10px;display:block;list-style:none;transition:.3s;color:#00aba5;font-family:Roboto,sans-serif;font-weight:600}.dropdown-container .dropdown-item .dropdown-link:hover{color:#fff;transition:.3s}.dropdown-menu{position:absolute;display:none;opacity:0;right:150px}.dropdown.open .dropdown-menu{display:block;opacity:1}.icon-image{cursor:pointer;margin-right:12px;display:inline-block;color:#fff;font-size:1.2em}`]
            },] },
];
/** @nocollapse */
NotificationDropdownComponent.ctorParameters = () => [];
NotificationDropdownComponent.propDecorators = {
    textNotificationDropdown: [{ type: Input }],
    titleNotificationDropdown: [{ type: Input }],
    typeNotificationDropdown: [{ type: Input }],
    classDropdown: [{ type: Input }]
};
function NotificationDropdownComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    NotificationDropdownComponent.prototype.textNotificationDropdown;
    /** @type {?} */
    NotificationDropdownComponent.prototype.titleNotificationDropdown;
    /** @type {?} */
    NotificationDropdownComponent.prototype.typeNotificationDropdown;
    /** @type {?} */
    NotificationDropdownComponent.prototype.classDropdown;
}
