// hcal/esm5/app/main/main.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
var MainComponent = /** @class */ (function () {
    function MainComponent() {
        this.title = 'System Name';
        this.credits = 'TECNOLOGIA DA INFORMAÇÃO - HCAL';
    }
    /**
     * @return {?}
     */
    MainComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    MainComponent.decorators = [
        { type: Component, args: [{
                    selector: 'hcal-main',
                    template: "<main class=\"container\">\n  <header class=\"top\">\n    <h2 class=\"title-header\">{{ title }}</h2>\n    <ng-content select=\"[nav-main]\"></ng-content>\n  </header>\n  <section class=\"medium\">\n    <ng-content select=\"[route]\"></ng-content>\n  </section>\n  <footer class=\"bottom\">\n    <p id=\"footer-ti\">{{ credits }}</p>\n  </footer>\n</main>\n",
                    styles: [".container{position:relative;width:calc(100% - 140px);height:100%;float:left;display:block;margin-left:140px;z-index:1;box-shadow:-5px 0 15px -7px;border-top-left-radius:10px;border-bottom-left-radius:10px}.container .top{width:100%;height:90px;background:#4dc2b3;border-top-left-radius:10px}.container .top .title-header{float:left;margin-top:29px;margin-left:25px;font-family:Roboto,serif;color:#fff;font-weight:500;cursor:default;text-transform:uppercase}.medium{height:calc(100% - 150px);padding:18px 50px;background:#f2f2f2;border-bottom-left-radius:20px}.bottom{width:calc(100% - 140px);height:25px;position:fixed;bottom:0;background:#4dc2b3;border-bottom-left-radius:10px}#footer-ti{margin-top:4px;text-align:center;font-family:Roboto,serif;color:#fff;font-size:13px;letter-spacing:3px;font-weight:500;cursor:default}"]
                },] },
    ];
    /** @nocollapse */
    MainComponent.ctorParameters = function () { return []; };
    MainComponent.propDecorators = {
        title: [{ type: Input }]
    };
    return MainComponent;
}());
export { MainComponent };
function MainComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    MainComponent.prototype.title;
    /** @type {?} */
    MainComponent.prototype.credits;
}
