// hcal/esm5/app/main/nav/nav.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
var NavComponent = /** @class */ (function () {
    function NavComponent() {
        this.hover = false;
        this.hiddenCog = false;
    }
    /**
     * @return {?}
     */
    NavComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    NavComponent.decorators = [
        { type: Component, args: [{
                    selector: 'hcal-nav',
                    template: "<nav class=\"menu-container main-menu\" main-menu>\n  <ul class=\"menu-content\">\n    <li class=\"menu-item\">\n      <ng-content select=\"[dropdown]\"></ng-content>\n    </li>\n    <li class=\"menu-item\" [hidden]=\"hiddenCog\">\n      <a routerLink=\"\" class=\"icon\">\n        <fa-icon class=\"icon-image cogs-icon\"\n                 [icon]=\"['fas', 'cog']\"\n                 (mouseover)=\"hover=true\"\n                 (mouseleave)=\"hover=false\"\n                 [spin]=\"hover\"></fa-icon>\n      </a>\n    </li>\n    <li class=\"menu-item\">\n      <a routerLink=\"\" class=\"icon\">\n        <fa-icon [icon]=\"['fas', 'sign-out-alt']\" class=\"icon-image\"></fa-icon>\n      </a>\n    </li>\n  </ul>\n</nav>\n",
                    styles: [".menu-container{margin-top:4px;margin-right:3%;float:right}.menu-container .menu-content{margin-top:3px;width:auto;padding:0;float:left}.menu-container .menu-content .menu-item{float:left;margin-top:27px;margin-right:15px}.icon-image{margin-right:12px;display:inline-block;color:#fff;font-size:1.2em}"]
                },] },
    ];
    /** @nocollapse */
    NavComponent.ctorParameters = function () { return []; };
    NavComponent.propDecorators = {
        hiddenCog: [{ type: Input }]
    };
    return NavComponent;
}());
export { NavComponent };
function NavComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    NavComponent.prototype.hover;
    /** @type {?} */
    NavComponent.prototype.hiddenCog;
}
