// hcal/fesm2015/hcal.js
import { Component, Input, Directive, ElementRef, HostListener, Renderer2, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
import { ChartModule, Chart } from 'angular-highcharts';
import { NgxPaginationModule } from 'ngx-pagination';
import { library } from '@fortawesome/fontawesome-svg-core';
import { fas } from '@fortawesome/free-solid-svg-icons';
import { far } from '@fortawesome/free-regular-svg-icons';
import { DropdownModule } from 'angular-custom-dropdown';
import { FormsModule } from '@angular/forms';
import { DualListComponent, AngularDualListBoxModule } from 'angular-dual-listbox';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class AsideBarComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
AsideBarComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-aside-bar',
                template: `<main class="container">
  <header class="top">
    <img id="logo" [src]="urlLogo" alt="Logo - Hospital do Coração de Alagoas">
  </header>
  <div class="menu-container">
    <ul class="menu-content">
      <ng-content select="[main-menu]"></ng-content>
    </ul>
  </div>
  <footer class="bottom"></footer>
</main>
`,
                styles: [`.container{position:fixed;width:150px;height:100%;float:left;z-index:initial;background-color:#f2f2f2}.container .top{width:100%;height:90px;background:#00aba5}.container .top #logo{width:70%;margin-top:5px;margin-left:16px;float:left}.container .menu-container{height:80%;z-index:5;position:relative}.container .menu-container .menu-content{position:relative;width:100%;margin:0;padding:0}.container .bottom{width:calc(100% - 140px);height:25px;position:fixed;bottom:0;background:#00aba5}`]
            },] },
];
/** @nocollapse */
AsideBarComponent.ctorParameters = () => [];
AsideBarComponent.propDecorators = {
    urlLogo: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class TooltipDirective {
    /**
     * @param {?} _elementRef
     * @param {?} _renderRef
     */
    constructor(_elementRef, _renderRef) {
        this._elementRef = _elementRef;
        this._renderRef = _renderRef;
        this.delay = 1000;
        this.isLabel = true;
        this.element = this._renderRef.selectRootElement(this._elementRef.nativeElement);
    }
    /**
     * @param {?} position
     * @return {?}
     */
    create(position) {
        this.parent = this._renderRef.parentNode(this.element);
        this.tooltipSpan = this._renderRef.createElement('span');
        this.tooltipText = this._renderRef.createText(this.tooltipHcal);
        this._renderRef.appendChild(this.tooltipSpan, this.tooltipText);
        switch (position) {
            case 'top':
                this.insertBefore();
                break;
            case 'bottom':
                this.insertAfter();
                break;
            default:
                this.insertAfter();
                break;
        }
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.create(this.position);
        this._renderRef.addClass(this.tooltipSpan, 'off-tooltipHcal');
    }
    /**
     * @param {?} delay
     * @param {?} _renderRef
     * @param {?} tooltipSpan
     * @param {?} position
     * @param {?} element
     * @param {?} isLabel
     * @return {?}
     */
    setTooltipInterval(delay, _renderRef, tooltipSpan, position, element, isLabel) {
        this.onDelay = setTimeout(function () {
            _renderRef.removeClass(tooltipSpan, 'off-tooltipHcal');
            _renderRef.addClass(tooltipSpan, 'on-tooltipHcal');
            _renderRef.addClass(tooltipSpan, position);
            /** @type {?} */
            let marginTop = '';
            if (isLabel) {
                marginTop = `calc(75% - ${element.offsetHeight / 2}px)`;
            }
            else {
                marginTop = `calc(50% - ${element.offsetHeight / 2}px)`;
            }
            /** @type {?} */
            const marginLeft = `calc(50% - ${tooltipSpan.offsetWidth / 2}px`;
            if (position === 'right' || position === 'left') {
                _renderRef.setStyle(tooltipSpan, 'top', marginTop);
                console.log(tooltipSpan.offsetHeight);
            }
            else {
                _renderRef.setStyle(tooltipSpan, 'left', marginLeft);
            }
        }, delay, false);
    }
    /**
     * @return {?}
     */
    onMouseOver() {
        this.setTooltipInterval(this.delay, this._renderRef, this.tooltipSpan, this.position, this.element, this.isLabel);
    }
    /**
     * @return {?}
     */
    outMouseOver() {
        clearTimeout(this.onDelay);
        this._renderRef.removeClass(this.tooltipSpan, this.position);
        this._renderRef.removeClass(this.tooltipSpan, 'on-tooltipHcal');
        this._renderRef.addClass(this.tooltipSpan, 'off-tooltipHcal');
    }
    /**
     * @return {?}
     */
    insertBefore() {
        this._renderRef.insertBefore(this.parent, this.tooltipSpan, this.element);
    }
    /**
     * @return {?}
     */
    insertAfter() {
        this._renderRef.appendChild(this.parent, this.tooltipSpan);
    }
}
TooltipDirective.decorators = [
    { type: Directive, args: [{
                selector: '[tooltipHcal]'
            },] },
];
/** @nocollapse */
TooltipDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
TooltipDirective.propDecorators = {
    tooltipHcal: [{ type: Input }],
    position: [{ type: Input }],
    delay: [{ type: Input }],
    isLabel: [{ type: Input }],
    onMouseOver: [{ type: HostListener, args: ['mouseover',] }],
    outMouseOver: [{ type: HostListener, args: ['mouseleave',] }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
library.add(fas, far);
class HcalSharedModule {
}
HcalSharedModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    TooltipDirective
                ],
                exports: [
                    TooltipDirective,
                    DropdownModule,
                    FontAwesomeModule,
                    RouterModule,
                    ChartModule,
                    NgxPaginationModule
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class LinkComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
LinkComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-link',
                template: `<li class="menu-item" routerLinkActive="active">
  <a [routerLink]="routerLinkName" class="full-link">
    <fa-icon [icon]="['fas', iconLink]" class="icon-aside"></fa-icon>
    <h3 class="title-link">{{ titleLink }}</h3>
  </a>
</li>
`,
                styles: [`.menu-item{margin-top:8px;margin-bottom:-3px;height:auto}.menu-item :hover{background-color:rgba(0,171,165,.45);transition:.7s}.menu-item .full-link{display:block;margin-left:-7.5px;text-decoration:none;text-align:center;font-family:"Roboto Medium",sans-serif;color:#00aba5}.menu-item .full-link:hover{color:#fff;background:rgba(0,171,165,.45)}.menu-item .full-link .icon-aside,.menu-item .full-link .title-link{display:block}.menu-item .full-link .icon-aside:hover,.menu-item .full-link .title-link:hover{background:0 0}.menu-item .full-link .icon-aside{margin-bottom:-7px;font-size:3em}.menu-item .full-link .title-link{padding-bottom:5px;width:auto;font-size:.9em;letter-spacing:.9px;font-weight:400}.active{background:rgba(0,171,165,.45)}.active .full-link{color:#fff}`]
            },] },
];
/** @nocollapse */
LinkComponent.ctorParameters = () => [];
LinkComponent.propDecorators = {
    titleLink: [{ type: Input }],
    routerLinkName: [{ type: Input }],
    iconLink: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class AsideBarModule {
}
AsideBarModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    HcalSharedModule
                ],
                declarations: [
                    AsideBarComponent,
                    LinkComponent
                ],
                exports: [
                    AsideBarComponent,
                    LinkComponent
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class MainComponent {
    constructor() {
        this.title = 'System Name';
        this.credits = 'TECNOLOGIA DA INFORMAÇÃO - HCAL';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
MainComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-main',
                template: `<main class="container">
  <header class="top">
    <h2 class="title-header">{{ title }}</h2>
    <ng-content select="[nav-main]"></ng-content>
  </header>
  <section class="medium">
    <ng-content select="[route]"></ng-content>
  </section>
  <footer class="bottom">
    <p id="footer-ti">{{ credits }}</p>
  </footer>
</main>
`,
                styles: [`.container{position:relative;width:calc(100% - 140px);height:100%;float:left;display:block;margin-left:140px;z-index:1;box-shadow:-5px 0 15px -7px;border-top-left-radius:10px;border-bottom-left-radius:10px}.container .top{width:100%;height:90px;background:#4dc2b3;border-top-left-radius:10px}.container .top .title-header{float:left;margin-top:29px;margin-left:25px;font-family:Roboto,serif;color:#fff;font-weight:500;cursor:default;text-transform:uppercase}.medium{height:calc(100% - 150px);padding:18px 50px;background:#f2f2f2;border-bottom-left-radius:20px}.bottom{width:calc(100% - 140px);height:25px;position:fixed;bottom:0;background:#4dc2b3;border-bottom-left-radius:10px}#footer-ti{margin-top:4px;text-align:center;font-family:Roboto,serif;color:#fff;font-size:13px;letter-spacing:3px;font-weight:500;cursor:default}`]
            },] },
];
/** @nocollapse */
MainComponent.ctorParameters = () => [];
MainComponent.propDecorators = {
    title: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NavComponent {
    constructor() {
        this.hover = false;
        this.hiddenCog = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
NavComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-nav',
                template: `<nav class="menu-container main-menu" main-menu>
  <ul class="menu-content">
    <li class="menu-item">
      <ng-content select="[dropdown]"></ng-content>
    </li>
    <li class="menu-item" [hidden]="hiddenCog">
      <a routerLink="" class="icon">
        <fa-icon class="icon-image cogs-icon"
                 [icon]="['fas', 'cog']"
                 (mouseover)="hover=true"
                 (mouseleave)="hover=false"
                 [spin]="hover"></fa-icon>
      </a>
    </li>
    <li class="menu-item">
      <a routerLink="" class="icon">
        <fa-icon [icon]="['fas', 'sign-out-alt']" class="icon-image"></fa-icon>
      </a>
    </li>
  </ul>
</nav>
`,
                styles: [`.menu-container{margin-top:4px;margin-right:3%;float:right}.menu-container .menu-content{margin-top:3px;width:auto;padding:0;float:left}.menu-container .menu-content .menu-item{float:left;margin-top:27px;margin-right:15px}.icon-image{margin-right:12px;display:inline-block;color:#fff;font-size:1.2em}`]
            },] },
];
/** @nocollapse */
NavComponent.ctorParameters = () => [];
NavComponent.propDecorators = {
    hiddenCog: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NavModule {
}
NavModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    HcalSharedModule
                ],
                declarations: [
                    NavComponent
                ],
                exports: [
                    NavComponent
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class MainModule {
}
MainModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    HcalSharedModule,
                    NavModule
                ],
                declarations: [
                    MainComponent
                ],
                exports: [
                    MainComponent,
                    NavModule
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NotificationComponent {
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
NotificationComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-notification',
                template: `<hcal-notification-content contentSize="col-8" notificationType="success"
                           notificationTitle="Notification Title" notificationText="Notification Content"></hcal-notification-content>
`,
                styles: [`.notification-head{float:left;height:80px;border-top-left-radius:15px}.notification-head .h3{padding:30px 10px;font-family:Poppins,sans-serif;color:#fff}.notification-body{height:auto;border-bottom-right-radius:15px}.notification-body .text{margin-top:15px;padding:10px;line-height:30px;font-family:Poppins,sans-serif;text-align:justify;color:#fff}`]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NotificationContentComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
NotificationContentComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-notification-content',
                template: `<main [class]="'content-notification float-left '+contentSize">
  <div [class]="'col-12 float-left notification-head bg-'+notificationType">
    <h3 class="h3">{{ notificationTitle }}</h3>
  </div>
  <div [class]="'col-12 offset-0 notification-body float-left bg-'+notificationType">
    <p class="text">{{ notificationText }}</p>
    <div class="col-12 float-right">
      <button [class]="'button col-2 float-right button-'+notificationType">Ok</button>
    </div>
  </div>
</main>
`,
                styles: [`.notification-head{float:left;height:80px;border-top-left-radius:15px}.notification-head .h3{padding:30px 10px;font-family:Poppins,sans-serif;color:#fff}.notification-body{height:auto;border-bottom-right-radius:15px}.notification-body .text{margin-top:15px;padding:10px;line-height:30px;font-family:Poppins,sans-serif;text-align:justify;color:#fff}.notification-body .button{margin-top:100px;margin-bottom:10px;margin-right:10px}`]
            },] },
];
/** @nocollapse */
NotificationContentComponent.ctorParameters = () => [];
NotificationContentComponent.propDecorators = {
    notificationText: [{ type: Input }],
    notificationTitle: [{ type: Input }],
    notificationType: [{ type: Input }],
    contentSize: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NotificationModule {
}
NotificationModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    HcalSharedModule
                ],
                declarations: [
                    NotificationComponent,
                    NotificationContentComponent
                ],
                exports: [
                    NotificationComponent,
                    NotificationContentComponent
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NotificationDropdownComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
NotificationDropdownComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-notification-dropdown',
                template: `<div dropdown>
  <ng-content select="[dropdownToggle]"></ng-content>
  <ul class="dropdown-menu notification-dropdown" dropdownMenu>
    <li class="notification-item">
      <div [class]="'container-notification-icon float-left bg-'+typeNotificationDropdown">
        <fa-icon [icon]="['fas', 'check']" class="notification-icon"></fa-icon>
      </div>
      <div [class]="'container-notification-content float-left bg-light-'+typeNotificationDropdown">
        <h3 class="notification-title">{{ titleNotificationDropdown }}</h3>
        <p class="content-notification">{{ textNotificationDropdown }}</p>
        <div class="notification-content-action-dropdown float-right">
          <button [class]="'button notification-action-dropdown button-'+typeNotificationDropdown">Read</button>
          <button [class]="'button notification-action-dropdown button-'+typeNotificationDropdown">Ok</button>
        </div>
      </div>
    </li>
  </ul>
</div>
`,
                styles: [`.notification-dropdown{width:400px;display:inline-block;float:left;height:auto;padding-top:10px;position:absolute;border-radius:10px;cursor:pointer;box-shadow:2px 2px 15px #999;transition:.3s;background:#fff}.notification-dropdown .notification-item{margin-bottom:10px;height:auto;float:left;border-radius:10px}.notification-dropdown .notification-item:hover{box-shadow:2px 2px 15px #999;transition:.3s}.notification-dropdown .notification-item .container-notification-icon{float:left;display:inline-block;min-width:80px;padding:20px;text-align:center;border-top-left-radius:10px;border-bottom-left-radius:10px}.notification-dropdown .notification-item .notification-icon{line-height:5px;font-size:4em;color:#fff}.notification-dropdown .notification-item .container-notification-content{float:left;display:inline-block;height:104px;min-width:250px;max-width:250px;border-top-right-radius:10px;border-bottom-right-radius:10px}.notification-dropdown .notification-item .container-notification-content .notification-title{padding-top:5px;padding-left:8px;margin-bottom:4px}.notification-dropdown .notification-item .container-notification-content .content-notification,.notification-dropdown .notification-item .container-notification-content .notification-title{color:#fff;font-family:Roboto,sans-serif;letter-spacing:1px}.notification-dropdown .notification-item .container-notification-content .content-notification{padding-left:10px;padding-right:10px;font-size:.9em;text-align:justify}.notification-dropdown .notification-item .container-notification-content .notification-content-action-dropdown{margin-top:3px;padding-left:8px}.notification-dropdown .notification-item .container-notification-content .notification-content-action-dropdown .notification-action-dropdown{border-radius:8px;margin-right:5px;font-weight:700;color:#fff;border-bottom:none}.dropdown-container{min-width:150px;max-width:370px;padding:20px;list-style-position:outside;border-radius:4px;box-shadow:2px 2px 15px #575756}.dropdown-container .dropdown-notification-item .dropdown-notification-icon{line-height:5px;font-size:4em;color:#fff}.dropdown-container .dropdown-item{margin:0;transition:.3s;display:block}.dropdown-container .dropdown-item:hover{background:#4dc2b3;box-shadow:2px 2px 15px #999;cursor:pointer;transition:.3s;border-radius:inherit}.dropdown-container .dropdown-item .dropdown-link{width:100%;padding:10px 0 10px 10px;display:block;list-style:none;transition:.3s;color:#00aba5;font-family:Roboto,sans-serif;font-weight:600}.dropdown-container .dropdown-item .dropdown-link:hover{color:#fff;transition:.3s}.dropdown-menu{position:absolute;display:none;opacity:0;right:150px}.dropdown.open .dropdown-menu{display:block;opacity:1}.icon-image{cursor:pointer;margin-right:12px;display:inline-block;color:#fff;font-size:1.2em}`]
            },] },
];
/** @nocollapse */
NotificationDropdownComponent.ctorParameters = () => [];
NotificationDropdownComponent.propDecorators = {
    textNotificationDropdown: [{ type: Input }],
    titleNotificationDropdown: [{ type: Input }],
    typeNotificationDropdown: [{ type: Input }],
    classDropdown: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NotificationDropdownModule {
}
NotificationDropdownModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    HcalSharedModule
                ],
                declarations: [NotificationDropdownComponent],
                exports: [NotificationDropdownComponent]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class TableComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
TableComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-table',
                template: `<div class="table-container">
  <table class="table-default">
    <caption class="tname">{{ tableName }}</caption>
    <thead class="thead-container">
    <tr class="trow-head">
      <ng-content select="[thead-content]"></ng-content>
    </tr>
    </thead>
    <tbody class="tbody-container">
      <ng-content></ng-content>
    </tbody>
  </table>
</div>
`,
                styles: [`.table-default{float:left;max-width:100%;min-width:100%;border-collapse:collapse;cursor:default}.table-default .tname{margin-bottom:10px;font-family:Roboto,sans-serif;text-align:left;font-size:1em;color:#2b517e}.table-default .thead-container{box-shadow:0 5px 12px -7px #000}.table-default .thead-default{padding:10px 40px;color:#fff;background:#2b517e;font-family:Roboto,sans-serif;text-transform:uppercase}.table-default .thead-default:first-child{border-top-left-radius:8px}.table-default .thead-default:last-child{border-top-right-radius:8px}.table-default .tbody-container .trow-content{color:#2b517e;font-family:Roboto,sans-serif;font-weight:300;text-align:center}.table-default .tbody-container .trow-content:nth-child(even){background:rgba(41,81,126,.1)}.table-default .tbody-container .trow-content:hover{cursor:pointer;background:rgba(41,81,126,.2);transition:.5s}.table-default .tbody-container .trow-content:last-child .tdata-default:first-child{border-bottom-left-radius:8px}.table-default .tbody-container .trow-content:last-child .tdata-default:last-child{border-bottom-right-radius:8px}.table-default .icon-table{color:#fff;font-size:1.2em}.table-default,.tdata-default,.thead-default,.trow-content{padding:10px}`]
            },] },
];
/** @nocollapse */
TableComponent.ctorParameters = () => [];
TableComponent.propDecorators = {
    tableName: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class TableModule {
}
TableModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    HcalSharedModule
                ],
                declarations: [
                    TableComponent
                ],
                exports: [
                    TableComponent
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class LoginComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
LoginComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-login',
                template: `<main class="container col-12">
  <div class="menu-container col-3">
    <div class="col-10" id="img-container">
      <img src="../../assets/img/heart.png" alt="">
    </div>
    <div class="menu-content-top col-12">
      <div class="system-info-container">
        <h1 id="system-info">HCAL</h1>
        <h4 id="system-name">Sistema</h4>
      </div>
    </div>
    <div class="menu-content-bottom col-12">
      <form class="teste col-10">
        <div class="input-container col-11">
          <fa-icon class="input-icon" [icon]="['fas','user']"></fa-icon>
          <input [position]="'top'" [delay]="5000" [tooltipHcal]="'Seu login ex: joaquim.jose'" type="text"
                 class="input-control" placeholder="login">
        </div>
        <div class="input-container col-11">
          <fa-icon class="input-icon" [icon]="['fas','lock']"></fa-icon>
            <input [position]="'left'" [delay]="5000" [tooltipHcal]="'A mesma senha que você usa para entrar no computador '" type="password" class="input-control" placeholder="senha">
        </div>

        <div class="position-absolute col-5">
          <div [position]="'bottom'" [tooltipHcal]="'Dica: é a mesma que você utiliza para logar no computador nome.sobrenome'"
                id="forget-pwd"> Esqueceu a senha?</div>
        </div>
        <button class="button-login">Entrar</button>
      </form>
    </div>
  </div>
</main>

`,
                styles: [`.container{position:relative;height:100%;background:#f2f2f2}.container .menu-container{position:absolute;top:50%;left:50%;margin:-200px 0 0 -220px;width:400px;height:440px;border-radius:0 50px 20px 100px;box-shadow:-5px 0 15px -7px,5px 0 15px -7px}.container .menu-container #img-container{position:absolute;margin-top:110px;margin-left:50px}.container .menu-container .menu-content-top{height:45%;display:block;border-top-right-radius:50px;border-top-left-radius:0;background-color:#4dc2b3}.container .menu-container .menu-content-top .system-info-container{float:right;margin-right:20px;margin-top:5px;font-family:"Roboto Black",sans-serif;color:#fff}.container .menu-container .menu-content-top .system-info-container #system-info{margin:0;padding:0;height:34px}.container .menu-container .menu-content-top .system-info-container #system-name{float:right;font-family:"Roboto Light",sans-serif;font-size:12px;letter-spacing:1px}.container .menu-container .menu-content-bottom{padding-top:40px;max-height:202px;height:60%;display:block;background-color:#fff;border-radius:0 0 20px 100px}.container .menu-container .menu-content-bottom #forget-pwd{display:inline;padding:1px;margin:0 20px;font-family:"Roboto Light",sans-serif;font-size:13px;letter-spacing:1px}.container .menu-container .menu-content-bottom .button-login{position:relative;padding:9px 30px;float:right;margin-top:3px;margin-right:20px;color:#fff;font-size:15px;font-family:"Roboto Black",sans-serif;font-weight:600;letter-spacing:1px;border-radius:50px;border:none;background-color:#2b517e}.container .menu-container .menu-content-bottom .button-login:hover{cursor:pointer;opacity:.9;transition:3s}.container .menu-container .menu-content-bottom .input-control{padding:7px 0;margin:5px auto;width:100%;font-family:"Roboto Bold",sans-serif;font-size:18px;color:#fff;text-align:center;border-radius:100px;background-color:#4dc2b3;border:none;transition:background-color 1s}.container .menu-container .menu-content-bottom .input-control::-webkit-input-placeholder{font-size:20px;font-family:"Roboto Bold",sans-serif}.container .menu-container .menu-content-bottom .input-control:-ms-input-placeholder{font-size:20px;font-family:"Roboto Bold",sans-serif}.container .menu-container .menu-content-bottom .input-control::-ms-input-placeholder{font-size:20px;font-family:"Roboto Bold",sans-serif}.container .menu-container .menu-content-bottom .input-control::placeholder{font-size:20px;font-family:"Roboto Bold",sans-serif}.container .menu-container .menu-content-bottom .input-container{position:relative}.container .menu-container .menu-content-bottom .input-container .input-icon{position:absolute;top:13px;left:13px;color:#fff}`]
            },] },
];
/** @nocollapse */
LoginComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class LoginModule {
}
LoginModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    HcalSharedModule,
                    FontAwesomeModule
                ],
                declarations: [
                    LoginComponent
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class ReportComponent {
    constructor() {
        this.chart = new Chart({
            chart: {
                type: 'pie'
            },
            title: {
                text: 'Graph',
            },
            credits: {
                enabled: false
            },
            series: [{
                    name: 'Pie 1',
                    data: [1, 2, 3]
                }]
        });
        this.componentName = this.constructor.name;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
ReportComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-report',
                template: `<h3 class="title-component">{{ componentName }}</h3>
<div class="content graph-container col-4 float-left">
  <div [chart]="chart" class="col-12 chart-content float-left"></div>
</div>
<div class="col-6 offset-1 float-left">
  <ng-content select="[table-report]"></ng-content>
</div>
`,
                styles: [`.graph-container{height:calc(100% - 140px)!important}`]
            },] },
];
/** @nocollapse */
ReportComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class ReportModule {
}
ReportModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    HcalSharedModule,
                    TableModule
                ],
                declarations: [
                    ReportComponent
                ],
                exports: [
                    ReportComponent
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class InputBaseComponent {
    constructor() {
        this.elementType = 'text';
        this.readonly = false;
        this.elementPlaceholder = '';
        this.enableTooltip = true;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
InputBaseComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-input-base',
                template: `<div [class]="'form-group position-relative '+contentStyle">
  <label [attr.for]="elementId" class="control-label">{{ elementLabel }}</label>
  <input [type]="elementType" [class]="className" name="elementName" [id]="elementId" [placeholder]="elementPlaceholder" [position]="placementTooltip"
         [tooltipHcal]="textTooltip" [delay]="delay" *ngIf="enableTooltip" ngModel [required]="enableRequired">
  <input *ngIf="!enableTooltip" [type]="elementType" [class]="className" ngModel name="elementName" [id]="elementId" [placeholder]="elementPlaceholder"
         [required]="enableRequired"
         [(ngModel)]="elementNgModel">
</div>
`,
                styles: [``]
            },] },
];
/** @nocollapse */
InputBaseComponent.ctorParameters = () => [];
InputBaseComponent.propDecorators = {
    elementId: [{ type: Input }],
    elementType: [{ type: Input }],
    elementLabel: [{ type: Input }],
    readonly: [{ type: Input }],
    maxLength: [{ type: Input }],
    elementNgModel: [{ type: Input }],
    elementPlaceholder: [{ type: Input }],
    elementName: [{ type: Input }],
    className: [{ type: Input }],
    textTooltip: [{ type: Input }],
    contentStyle: [{ type: Input }],
    placementTooltip: [{ type: Input }],
    delay: [{ type: Input }],
    enableTooltip: [{ type: Input }],
    enableRequired: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class InputBaseModule {
}
InputBaseModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    FormsModule,
                    HcalSharedModule
                ],
                declarations: [
                    InputBaseComponent
                ],
                exports: [
                    InputBaseComponent
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class SelectBaseComponent extends InputBaseComponent {
    constructor() {
        super();
        this.options = [];
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
SelectBaseComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-select-base',
                template: `<div [class]="'form-group '+contentStyle">
  <label [attr.for]="elementId" class="control-label">{{ elementLabel }}</label>
  <select [name]="elementName" [id]="elementId" [class]="className">
    <option value="" *ngFor="let option of options">{{ option.object.nome }}</option>
  </select>
</div>
`,
                styles: [``]
            },] },
];
/** @nocollapse */
SelectBaseComponent.ctorParameters = () => [];
SelectBaseComponent.propDecorators = {
    options: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class SelectBaseModule {
}
SelectBaseModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    SelectBaseComponent
                ],
                exports: [
                    SelectBaseComponent
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class CustomDualListComponent extends DualListComponent {
    constructor() {
        super(...arguments);
        this.format = {
            add: '', remove: '', all: 'Todos', none: 'Nenhum',
            direction: DualListComponent.LTR, draggable: true, locale: 'br'
        };
        this.fieldDestination = [];
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
CustomDualListComponent.decorators = [
    { type: Component, args: [{
                selector: 'hcal-custom-dual-list',
                template: `<dual-list [source]="data" [destination]="fieldDestination" [format]="format"></dual-list>
`,
                styles: [``]
            },] },
];
CustomDualListComponent.propDecorators = {
    data: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class CustomDualListModule {
}
CustomDualListModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    HcalSharedModule,
                    AngularDualListBoxModule
                ],
                declarations: [
                    CustomDualListComponent
                ],
                exports: [
                    CustomDualListComponent
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { AsideBarModule, MainModule, HcalSharedModule, NavModule, NotificationModule, NotificationDropdownModule, TableModule, LoginModule, ReportModule, InputBaseModule, SelectBaseModule, CustomDualListModule, TooltipDirective, AsideBarComponent as ɵa, LinkComponent as ɵb, CustomDualListComponent as ɵm, InputBaseComponent as ɵk, SelectBaseComponent as ɵl, LoginComponent as ɵi, MainComponent as ɵd, NavComponent as ɵc, NotificationContentComponent as ɵf, NotificationDropdownComponent as ɵg, NotificationComponent as ɵe, ReportComponent as ɵj, TableComponent as ɵh };
