// hcal/esm5/app/form/custom-dual-list/custom-dual-list.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
import { DualListComponent } from 'angular-dual-listbox';
var CustomDualListComponent = /** @class */ (function (_super) {
    tslib_1.__extends(CustomDualListComponent, _super);
    function CustomDualListComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.format = {
            add: '', remove: '', all: 'Todos', none: 'Nenhum',
            direction: DualListComponent.LTR, draggable: true, locale: 'br'
        };
        _this.fieldDestination = [];
        return _this;
    }
    /**
     * @return {?}
     */
    CustomDualListComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    CustomDualListComponent.decorators = [
        { type: Component, args: [{
                    selector: 'hcal-custom-dual-list',
                    template: "<dual-list [source]=\"data\" [destination]=\"fieldDestination\" [format]=\"format\"></dual-list>\n",
                    styles: [""]
                },] },
    ];
    CustomDualListComponent.propDecorators = {
        data: [{ type: Input }]
    };
    return CustomDualListComponent;
}(DualListComponent));
export { CustomDualListComponent };
function CustomDualListComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    CustomDualListComponent.prototype.format;
    /** @type {?} */
    CustomDualListComponent.prototype.data;
    /** @type {?} */
    CustomDualListComponent.prototype.fieldDestination;
}
