// hcal/esm5/app/form/select-base/select-base.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
import { InputBaseComponent } from '../input-base/input-base.component';
var SelectBaseComponent = /** @class */ (function (_super) {
    tslib_1.__extends(SelectBaseComponent, _super);
    function SelectBaseComponent() {
        var _this = _super.call(this) || this;
        _this.options = [];
        return _this;
    }
    /**
     * @return {?}
     */
    SelectBaseComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    SelectBaseComponent.decorators = [
        { type: Component, args: [{
                    selector: 'hcal-select-base',
                    template: "<div [class]=\"'form-group '+contentStyle\">\n  <label [attr.for]=\"elementId\" class=\"control-label\">{{ elementLabel }}</label>\n  <select [name]=\"elementName\" [id]=\"elementId\" [class]=\"className\">\n    <option value=\"\" *ngFor=\"let option of options\">{{ option.object.nome }}</option>\n  </select>\n</div>\n",
                    styles: [""]
                },] },
    ];
    /** @nocollapse */
    SelectBaseComponent.ctorParameters = function () { return []; };
    SelectBaseComponent.propDecorators = {
        options: [{ type: Input }]
    };
    return SelectBaseComponent;
}(InputBaseComponent));
export { SelectBaseComponent };
function SelectBaseComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    SelectBaseComponent.prototype.options;
}
